File: Chapter 03/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter03/section_3_3_5_dip/after/Forward.java
package com.grokkingsoftwarearchitecture.chapter03.section_3_3_5_dip.after;

/**
 * A Low-Level Detail.
 * * ARCHITECTURE NOTE: The concrete implementation now depends on the abstraction 
 * (Player). The flow of dependency has been strictly inverted.
 */
public class Forward implements Player {
    public void performAction() {
        System.out.println("  [Action] Forward is attacking.");
    }
}

File: Chapter 03/C#/Section_3_3_5_DIP/After/Forward.cs
using System;

namespace Chapter03.DIP.After;
/// <summary>
/// A Low-Level Detail.
/// 
/// ARCHITECTURE NOTE: The concrete implementation now depends on the abstraction 
/// (IPlayer). The flow of dependency has been inverted.
/// </summary>
public class Forward : IPlayer
{
    public void PerformAction() => Console.WriteLine("  [Action] Forward is attacking.");
}